### DISCUSSION TITLE
Why Codespaces storage usage warning?

### DISCUSSION THREAD

**Discussion by Kebap** [community/community#39615] [Codespaces] [answered]
### Select Topic Area

Question

### Body

Hey! A few days ago, I have received an email warning me I had used 75% of included services for GitHub Codespaces Storage.
Then yesterday I received another email with the same warning at 90% now. There were some links to update the spending limit.
However, I didn't even use Codespaces in the meantime. Not to my knowledge at least.
How can this happen? How can I review how or where the storage is/was used exactly?
Maybe I can free up some storage now and reduce the 90% again?
Not sure why it even got so large or increased further the last days.
Thanks for your time and possibly some advice! :)

**Comment by ossanna16** [comment]
Hi there @Kebap and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Reply by jainam03** [reply]
I keep getting email form github support saying that I've used 90% of my Github Codespaces storage services....but when I login and visit the codespaces section it shows that there are no active codespaces....what's wrong exactly? 🤔
please help!

![image](https://github.com/community/community/assets/91124384/b07ab3e0-f9f9-4f0b-94a6-18d25cf6461b)

![image](https://github.com/community/community/assets/91124384/4617c605-da4e-4038-8637-85b1e416ad30)

**Reply by wmstack** [reply]
Same here, I am not aware of any active Codespaces I have but keep being told about some 90% limit even though I don't even remember using Github Codespaces anytime this month.

**Comment by dragontheory** [comment]
I am getting the same warning. Started last week. Both emails and now a message in VSCode itself that says...

"You've used 90% of included services for GitHub Codespaces Storage. To continue using Codespaces uninterrupted, update your spending limit." 

The repo I am editing in Codespaces is 13KB zipped when downloaded from GitHub. 

Here is the link.
[https://github.com/dragontheory/D7460N](https://github.com/dragontheory/D7460N)

Strange that there are no links or prompts to see how much space I have left or how I have used so much space? Just a link to pay for more space. 

Nowhere does Codespace show storage space usage. 

Please advise as to how this can be investigated or this error fixed. 

Thank you.

**Comment by bozdoz** [comment]
Same issue here. I do not use GitHub Codespaces, so why am I getting emails claiming "You've used 90% of included services"?

**Comment by louh** [comment, marked as answer]
I just ran into this issue, and based on this conversation [in this thread](https://github.com/orgs/community/discussions/39133), it seems this recently rolled out and it's buggy as hell. For one thing, old Codespaces are calculated as accumulating storage every single day, even if you just have one of them and never use it. So one old Codespace at 30 days a month is like saying you have 30 repos using Codespaces a month. It chews through the usage limit _incredibly_ quickly.

Here's what to do:
- See your list of Codespaces here: https://github.com/codespaces (they really need to link this somewhere that's easier to get to)
- Delete all Codespaces (probably one or two you tried, years ago)
- In a few days, double check the [usage report](https://docs.github.com/en/billing/managing-billing-for-github-codespaces/viewing-your-github-codespaces-usage) and make sure it has stopped accruing usage. The last date should be the day you deleted the codespaces.

**Reply by SpamixOfficial** [reply]
I have the same issue, but I don’t have any codespaces. How would I fix it in this case?

**Reply by skuchler** [reply]
Same issue for me. I don't have any codespaces. In the report I see "Create Codespaces Prebuilds" billed every day, but it's related to an old repository that I'm not working on.

**Reply by saru2020** [reply]
I'm getting these emails ('You've used 100% of included services for the saru2020 account') as well although I haven't used codespaces at all

**Reply by davismj** [reply]
Yikes. So glad I didn't give Github my billing info.

**Reply by mattpilott** [reply]
same boat, have no codespaces, cant see anywhere that could be contributing to this usage limit and yet its increasing, anyone solve this yet?

**Reply by davismj** [reply]
codespaces prebuild, on one of your repos

**Reply by mattpilott** [reply]
how do i find this though?

**Reply by soup-bowl** [reply]
You guys have codespaces that are years old?! I can barely have 1 or 2 codespaces lingering before I run out of storage.

**Reply by dragontheory** [reply]
Even after deleting the one 13k repo (so there is nothing in Codespaces at all) over 6 months ago , I'm still getting the same usage warnings and how I should "upgrade" my plan. Flushed cache and all, but it is still there. My projects are minimal and without external dependencies. Wish they would fix this. I would use Codespaces then.

**Reply by soup-bowl** [reply]
[Gitpod](https://www.gitpod.io) is a good alternative that doesn't set limits on storage, just on hours used.

**Reply by vasilvestre** [reply]
I don't have any codespace listed but i keep receiving alert because of the max usage.

**Reply by ErickRodrCodes** [reply]
> I don't have any codespace listed but i keep receiving alert because of the max usage.

On the same boat. I don't even have a codespace.

**Comment by TouchstoneTheDev** [comment]
i am using codespaces but storage is only 32 gb :(

**Reply by TouchstoneTheDev** [reply]
i want to build a private llm

**Reply by TouchstoneTheDev** [reply]
codespaces have good bandwidth :) but limited storage

**Reply by TouchstoneTheDev** [reply]
Hey @github thenks for some upgrade

**Comment by ErickRodrCodes** [comment]
Hello Github. I'm annoyed as hell as I don't have any codespaces, and I have 20GB used???

proof 1: list of codespaces:
![image](https://github.com/community/community/assets/1978642/1291d62f-91c5-4cc6-aeca-97fcffca1d57)


Proof 2: Where I have used 20GB of that??
![image](https://github.com/community/community/assets/1978642/03a4fc04-e485-4826-b28c-2a97f118b022)
![image](https://github.com/community/community/assets/1978642/66667b89-0930-4065-b30e-41c1750879a3)

Please Help!!!

**Comment by SpamixOfficial** [comment]
So, after digging around a little I can safely say that the warning I'm getting is from prebuild storage. From the screenshots I've seen from other people this also seems to be the case for them. 

Please GitHub, make your goddamn emails more clear about what you actually mean.

**Comment by SpamixOfficial** [comment]
> Hello Github. I'm annoyed as hell as I don't have any codespaces, and I have 20GB used???
> 
> 
> 
> proof 1: list of codespaces:
> 
> ![image](https://github.com/community/community/assets/1978642/1291d62f-91c5-4cc6-aeca-97fcffca1d57)
> 
> 
> 
> 
> 
> Proof 2: Where I have used 20GB of that??
> 
> ![image](https://github.com/community/community/assets/1978642/03a4fc04-e485-4826-b28c-2a97f118b022)
> 
> ![image](https://github.com/community/community/assets/1978642/66667b89-0930-4065-b30e-41c1750879a3)
> 
> 
> 
> Please Help!!!
> 
> 

See my answer above. Prebuild storage is basically the stuff you generate on builds with actions. Try auto-removing some of that!

**Reply by ErickRodrCodes** [reply]
> > Hello Github. I'm annoyed as hell as I don't have any codespaces, and I have 20GB used???
> > proof 1: list of codespaces:
> > ![image](https://private-user-images.githubusercontent.com/1978642/346144261-1291d62f-91c5-4cc6-aeca-97fcffca1d57.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA0MTQ0MTAsIm5iZiI6MTcyMDQxNDExMCwicGF0aCI6Ii8xOTc4NjQyLzM0NjE0NDI2MS0xMjkxZDYyZi05MWM1LTRjYzYtYWVjYS05N2ZjZmZjYTFkNTcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MDhUMDQ0ODMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGJlMmMxNjU0ZWVjYTk5MGZhNjliNzY3NTY0NWEyYWI0MWYzYzg5NTgxNDIyZTE1MjczM2Q1YTZlZjQ3YTAwYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.8Qr8DjrQI5T970k-bwXQWFn5UxFumNwBKG0ntRw_tgo)
> > Proof 2: Where I have used 20GB of that??
> > ![image](https://private-user-images.githubusercontent.com/1978642/346144525-03a4fc04-e485-4826-b28c-2a97f118b022.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA0MTQ0MTAsIm5iZiI6MTcyMDQxNDExMCwicGF0aCI6Ii8xOTc4NjQyLzM0NjE0NDUyNS0wM2E0ZmMwNC1lNDg1LTQ4MjYtYjI4Yy0yYTk3ZjExOGIwMjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MDhUMDQ0ODMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWMwNTdhMzAyMjY5NDUxM2UxYzU2Yjk4MTg0NmNiMjk5YjVhNGFjOGRjYWNjZmFlZTQ5NDhhOGNlMDk3N2IyZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.8ZlUKHiOeN_Kfo52Z7tuuwveuMJb7MZURQSVar-zkPM)
> > ![image](https://private-user-images.githubusercontent.com/1978642/346144645-66667b89-0930-4065-b30e-41c1750879a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA0MTQ0MTAsIm5iZiI6MTcyMDQxNDExMCwicGF0aCI6Ii8xOTc4NjQyLzM0NjE0NDY0NS02NjY2N2I4OS0wOTMwLTQwNjUtYjMwZS00MWMxNzUwODc5YTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MDhUMDQ0ODMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjhmMjI1OGNjMjBjNmQ4ZThlODYyNjQ1N2IwYTM4NjUwOTFlOTEzODQ0N2Q1NTZmZDFiMDU3MjYwYThiNDUyZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.h2vmaaR-7CVP-zX6CaZhyCJVZcNUF_RK-2aB4quBfyo)
> > Please Help!!!
> 
> See my answer above. Prebuild storage is basically the stuff you generate on builds with actions. Try auto-removing some of that!

I have checked up my repos, and I find almost impossible I have ever generated any action at all.

**Comment by audacioustux** [comment]
Facing similar issue... I can't find any prebuild configured on any of my repos but prebuild still taking storage. 
<img width="943" alt="Screenshot 2024-07-28 at 12 46 48 AM" src="https://github.com/user-attachments/assets/72c3a1fa-1c02-46af-a43d-8ef37b76ce64">

**Comment by RemyJrd** [comment]
I'm so close to close my github account because of that, those dudes are blind, or idiot ?

**Comment by hugmouse** [comment]
@copilot pls fix this

<img width="350" height="937" alt="Screenshot of Github Codespace emails" src="https://github.com/user-attachments/assets/97f84c32-fbea-4698-8c0d-db746cb0a4ef" />

**Comment by keioro** [comment]
This can happen because codespaces use storage even when you aren't actively using. Check metered usage settings

**Comment by vitorcgo** [comment]
Interessante

**Comment by hugmouse** [comment]
Ok I was able to find which repos were doing that through gh cli. Couldn't find them via github's frontend.

Here's bash script. Have `gh`, `grep`, `sort` installed.

```bash
#!/usr/bin/env bash
set -uo pipefail

command -v gh >/dev/null || { echo "gh is required"; exit 1; }
gh auth status >/dev/null 2>&1 || { echo "run: gh auth login"; exit 1; }

ME=$(gh api /user -q .login)
echo "account: $ME"

echo
echo "== billing: every codespaces SKUs, quantity =="
gh api "/users/$ME/settings/billing/usage" \
  --jq '.usageItems[] | select(.product=="codespaces") | "\(.date[0:7])  \(.sku)  \(.quantity)"'

echo
echo "== codespaces that still exist =="
gh codespace list --limit 100 || true

echo
echo "== repos with a prebuild config =="
gh api -X GET /search/code -f "q=filename:devcontainer.json user:$ME" \
  --jq '.items[].repository.full_name' 2>/dev/null | sort -u |
  while read -r r; do
    gh workflow list -R "$r" 2>/dev/null | grep -qi prebuild && echo "$r"
  done
  ```
  
  It will output something like this:
  
  ```bash
  account: hugmouse

== billing: codespaces SKUs, quantity ==
2026-02  Codespaces compute 2-core  4.73009
2026-02  Codespaces compute 8-core  0.63603
2026-02  Codespaces prebuild storage  7.939131326
2026-02  Codespaces storage  0.378277327
2026-03  Codespaces compute 2-core  0.29789
2026-03  Codespaces prebuild storage  14.665550297
2026-03  Codespaces storage  0.334466085
2026-04  Codespaces compute 2-core  1.09609
2026-04  Codespaces prebuild storage  14.510251098
2026-04  Codespaces storage  0.101495822
2026-05  Codespaces prebuild storage  15.000014309
2026-06  Codespaces prebuild storage  15.005227268
2026-07  Codespaces prebuild storage  15.000011207
2026-08  Codespaces prebuild storage  15.000011207
2026-09  Codespaces prebuild storage  15.000009143

== codespaces that still exist ==
no codespaces found

== repos with a prebuild config ==
hugmouse/dicks
hugmouse/balls
```
